lib/_stream_writable.js

```diff
@@ -586,8 +587,17 @@ Writable.prototype.end = function(chunk, encoding, cb) {
     encoding = null;
   }
 
-  if (chunk !== null && chunk !== undefined)
-    this.write(chunk, encoding);
+  if (typeof cb === 'function') {
+    cb = once(cb);
+  }
+
+  if (chunk != null) {
```

Comment:
It is possible to keep `chunk !== null && chunk !== undefined` would be just as good, `chunk != will` be implicitly type-transformed.